### DISCUSSION TITLE
What is the right way to undo an unplug?

### DISCUSSION THREAD

**Discussion by shripadk** [yarnpkg/berry#2614] [Q&A] [answered]
I had to unplug a dependency to debug it. Now that I figured out the issue with the dependency I want to revert back to the original state. So, how do I undo an unplug?

The [documentation](https://yarnpkg.com/cli/unplug#details) says:

> As such, to undo its effects, you'll need to revert the changes made to the manifest and run yarn install to apply the modification.

Which manifest file should I make these changes in?

**Comment by merceyz** [comment, marked as answer]
The sentence before the one you quoted specifies which manifest

> Running the command will set a persistent flag inside your top-level package.json, in the dependenciesMeta field

**Reply by shripadk** [reply]
I tried that. Even removed the `.yarn` folder and `.yarnrc.yml` file and reinstalled yarn from scratch. But the package is still unplugged in `.yarn/unplugged` folder. Which is what got me wondering if there was another manifest file the documentation was alluding to.

**Reply by shripadk** [reply]
Okay I understood now. I tried it with another package and it worked. I missed reading this part of the documentation: https://yarnpkg.com/advanced/lexicon/#unplugged-package

> Packages are unplugged in two cases: either explicitly when the unplugged field is set to true, or implicitly when the package lists postinstall scripts.

Now I know that the package was always unplugged regardless of whether I unplugged it or not. Sorry for the trouble. This makes everything clear to me.
